### PR TITLE
Hide placeholders behind fallback images on image load error

### LINKED ISSUE #10646: amp-image: Placeholder fails to hide when unsupported source is supplied
## What's the issue?

In our template we are using WebP and falling back to PNG as recommended by the documentation: https://www.ampproject.org/docs/reference/components/amp-img

We also have a custom placeholder which is visible when the image is loading. However on browsers that do not support WebP, the placeholder fails to disappear when the fallback image loads.

## How do we reproduce the issue?

Please see the attached [codepen](https://codepen.io/willdonohoe/pen/oeXYNx).

1) Open the codepen above in Chrome
2) Open dev tools and turn on network throttling
3) Refresh and notice the loading placeholder (Yellow square) appears while loading and disappears when the image appears
4) Open the same page in Safari (Or a browser which does not support WebP)
5) Notice that the placeholder remains on the screen when the fallback image is loaded.

Note: This issue is not visible when using opaque images, as the fallback image appears over the top of the loader. However our fallback is transparent and therefore the loader can be seen underneath.

## Expected behaviour

`amp-image` should hide the placeholder if the image is not supported.

### PR DESCRIPTION
**Pull request by torch2424**
closes #10646 

This will hide all of the child placeholder elements within an amp-image component, when the image fails to load. Please let me know if there is a case where this would lead to unexpected results. Also, let me know if you would like me to add the `test/manual` file I added for this fix

# Screenshot (Safari and Firefox)

screen shot 2017-07-28 at 1 32 10 pm

### CODE REVIEW

File: builtins/amp-img.js [hunk 1]
```diff
@@ -197,8 +199,22 @@ export class AmpImg extends BaseElement {
     this.getVsync().mutate(() => {
       this.img_.classList.add('i-amphtml-ghost');
       this.toggleFallback(true);
+      // Hide child placeholders, as browsers that don't support webp
+      // Would show the placeholder underneath a transparent fallback
+      this.hideChildPlaceholders_();
```
Comment (jridgewell) [inline builtins/amp-img.js]: Why aren't you just using `#togglePlaceholder`?

Comment (torch2424) [inline builtins/amp-img.js]: Oh I didn't see we had that, thanks! I'll do that now

**Comment by torch2424** [comment]
@jridgewell made your changes. Thanks for the catch!

File: builtins/amp-img.js [hunk 2]
```diff
@@ -15,6 +15,8 @@
  */
 
 import {BaseElement} from '../src/base-element';
+import {toggle} from '../src/style';
+import {toArray} from '../src/types';
```
Comment (jridgewell) [inline builtins/amp-img.js]: Unused imports.

Comment (torch2424) [inline builtins/amp-img.js]: My apologies, forgot to run lint.

File: builtins/amp-img.js [hunk 3]
```diff
@@ -197,6 +197,9 @@ export class AmpImg extends BaseElement {
     this.getVsync().mutate(() => {
       this.img_.classList.add('i-amphtml-ghost');
       this.toggleFallback(true);
+      // Hide placeholders, as browsers that don't support webp
+      // Would show the placeholder underneath a transparent fallback
+      this.togglePlaceholder(false);
```
Comment (dvoytenko) [inline builtins/amp-img.js]: @torch2424 Could you please add a test in a separate PR?

Comment (torch2424) [inline builtins/amp-img.js]: Sure! I'll make an issue for this, and assign myself to it.

**Pull request merged**